python/ray/tests/test_task_metrics.py

```diff
@@ -318,20 +324,45 @@ def test_task_retry(shutdown_only):
 
 ray.init("auto")
 
-@ray.remote(retry_exceptions=True)
+@ray.remote
+def sleep():
+    time.sleep(999)
+
+@ray.remote
+class Phaser:
+    def __init__(self):
+        self.i = 0
+
+    def inc(self):
+        self.i += 1
+        if self.i < 3:
+            raise ValueError("First two tries will fail")
+
+phaser = Phaser.remote()
+
+@ray.remote(retry_exceptions=True, max_retries=3)
 def f():
-    assert False
+    ray.get(phaser.inc.remote())
+    ray.get(sleep.remote())
 
 f.remote()
 time.sleep(999)
 """
 
     proc = run_string_as_driver_nonblocking(driver)
     expected = {
-        "FAILED": 1.0,  # Only recorded as finished once.
+        ("sleep", "RUNNING", "0"): 1.0,
+        ("f", "FAILED", "0"): 1.0,
+        ("f", "FAILED", "1"): 1.0,
+        ("f", "RUNNING_IN_RAY_GET", "1"): 1.0,
+        ("Phaser.__init__", "FINISHED", "0"): 1.0,
+        ("Phaser.inc", "FINISHED", "0"): 1.0,
```

Comment:
Maybe we should also test actor test retry case? 
<img width="638" alt="Screen Shot 2022-12-01 at 3 45 44 PM" src="https://user-images.githubusercontent.com/18510752/205182601-38cab9cc-a2ab-426c-877f-e69de62c0dc7.png">